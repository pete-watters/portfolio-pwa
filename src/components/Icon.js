import React from 'react';
import PropTypes from 'prop-types';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faTwitter, faLinkedin, faStackOverflow, faGithub, faMedium } from '@fortawesome/free-brands-svg-icons';

const ICON_TYPES = {
  twitter: faTwitter, 
  linkedIn: faLinkedin, 
  stackOverflow: faStackOverflow, 
  github: faGithub,
  medium: faMedium,
}
const Icon = ({ icon, href }) =>
  <a href={href}target="_blank" rel="noopener noreferrer">
    <FontAwesomeIcon icon={ICON_TYPES[icon]} />
  </a>;

Icon.propTypes = {
  icon: PropTypes.string.isRequired,
  href: PropTypes.string.isRequired,
};

export default Icon;