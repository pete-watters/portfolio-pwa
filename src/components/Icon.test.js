import React from 'react';
import { shallow } from 'enzyme';
import Icon from './Icon';

describe('Icon', () => {
  it('should render the component correctly', () => {
    expect(shallow(<Icon icon='faTwitter' href='#' onClick={null} />)).toMatchSnapshot();
  });
});
