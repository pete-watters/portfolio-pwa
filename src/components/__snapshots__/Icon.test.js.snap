// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should render the component correctly 1`] = `
<a
  aria-label="faTwitter"
  href="#"
  onClick={null}
  rel="noopener noreferrer"
  target="_blank"
  title="FaTwitter"
>
  <FontAwesomeIcon
    border={false}
    className=""
    fixedWidth={false}
    flip={null}
    icon={null}
    inverse={false}
    listItem={false}
    mask={null}
    pull={null}
    pulse={false}
    rotation={null}
    size={null}
    spin={false}
    swapOpacity={false}
    symbol={false}
    title=""
    transform={null}
  />
</a>
`;
